### DISCUSSION TITLE
Web issues with some of the later firmware

### DISCUSSION THREAD

**Discussion by ChadH360** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#672] [General]
I've been using the WiFi remote for some time now with Home Assistant - absolutely excellent project, I absolutely love it!
It's getting close to that time of year when I start using the hot tub again, so I thought I'd update it.
I was previously running firmware version 2023-03-27-2004 with no issues that I was aware of.

I initally updated to the latest version from the Master branch (this was  on 21st Feb 2024), and due to some issues which I'll go into a bit, I updated to the latest version from the Development_v4 branch. According to the web interface it was firmware version 2023-11-19-1330, but I still had the same issues.

Initially, I tried an OTA update with it still in the Hot Tub, and whilst this sort of worked, I lost the ability to manage the device, and also the Hot Tub itself stopped functioning, so I took it out and configured it from my computer.
It turns out that the Hot Tub stopped functioning as the Hardware Config was no longer correct, and although the device did remember the WiFi details, I couldn't access the web interface.

So, I connected up to my computer, erased the flash, flashed it with the new firmware, and then configured accordingly via the Web interface. Everything seemed fine.
I then put it back in to the Hot Tub (incidentally, a Maldives 2021 - 6 wire 2021 with hydrojets, using a home made build based on the V1 PCB)
Hot Tub appeared to work absolutely fine, MQTT into Home Assistant was absolutely fine, but for some reason, the Web Interface was no longer accessible.

I have a Unifi Mesh Wifi network made up of 3 x 'U6 Mesh' devices.
They are named Upstairs, Downstairs and Outside. Upstairs and Downstairs are wired into my network, Outside is wirelessly meshed to Downstairs.

After a lot of head scratching, erasing, reflashing, etc, etc I basically discovered that with the newer firmware, if it connects to Downstairs then everything works fine - hence why it worked fine when connected to my laptop inside the house.
But, as soon as it connected to Outside (which is where the Hot Tub is), then despite everything appearing to function normally from Home Assistant and the Hot Tub, the Web Interface wouldn't work.

The symptoms appear to be similar to discussions/466 and issues/504 whereby I can ping, but when trying to load the web page, the ping will timeout briefly and nothing will display in the browser (after some time, it may display some unformatted text from the main menu, but nothing that's usable).
I think both of these related to Unifi networks as well

However, by reverting back to firmware 2023-03-27-2004 (I still had this in my downloads from last year), everything is working normally.

Any ideas on what might be causing this? Is there anything I can do to help diagnose?

**Comment by visualapproach** [comment]
Hi, I know the websocket connection doesn't work across networks. But then it shouldn't work with the old fw either...

**Comment by ChadH360** [comment]
Ah, sorry I should have mentioned.
This is all on the same SSID/network - just multiple access points throughout the property. The module will connect to the nearest AP, so whilst I had it near my laptop it was connecting to Downstairs, and when I put it in my Hot Tub, it was connecting to Outside.
The only difference I can see between Outside and Downstairs is that Outside wirelessly meshes to Downstairs, whereas Downstairs is directly wired to the network. This is all handled by Unifi, though, and the ESP shouldn't be aware of any difference - worst case, the ping times might be marginally slower, but we are literally talking by about 3 or 4 ms.
The only other difference I can see is that Outside is on Channel 6 and Downstairs is on Channel 11, but I really can't see how that might make any difference.

I just wonder if the ESP is somehow sensitive to the Wireless Meshing that Unifi offers?
https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/issues/504 seems to have exactly the same symptoms, and they all seem to be using Unifi APs.
One person also mentioned that their AP switched from channel 6 to channel 1 and then it seemed OK - maybe it has some issue with Channel 6?!
I'll try forcing my AP to a different channel and see if it sorts the issue on the later firmware.

**Comment by ChadH360** [comment]
?!?!?!
Just forced my Outside AP from Channel 6 to Channel 1, and everything is now working with the latest firmware (2023-11-19-1330).
Why would it have an issue on Channel 6, but only with the later firmware??
Bizarre.
But if you read through https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/issues/504, a similar thing happened there when it moved from Channel 6 to Channel 1
Seems crazy!
Any thoughts?

Could it perhaps be a bug with the the library responsible for Wifi? I assume the later firmware may be using later libraries that could have some wierd bug in them? (Sorry, I'm not much of a programmer so not sure how accurate that assumption might be!)

**Reply by visualapproach** [reply]
Oh, that's weird. I have no idea why that is but good to know there's a workaround. Glad you found out.

**Reply by ChadH360** [reply]
So, a bit more testing - I've tried channels 1 through to 11.
Channels 5, 6 and 7 don't work.
When I say don't work, MQTT works fine, the Hot Tub works fine, and I can ping the IP address with no problems.
It just won't display the web interface. I can get the /dir/ page to load, but none of the main configuration pages will load properly. There is also some dropped pings when attempting to access the web pages.

So weird! I have definitely seen other Unifi users on here have a similar (most likely the same) issue, and presume maybe it is just a Unifi thing - it would be interesting to know how other users get on with channels 5, 6 and 7 on other router brands.
The channel issue applies to both the wired Downstairs AP, and the wireless Outside AP, but very weirdly not on earlier versions of the firmware.

So, if you're having trouble accessing the web interface, switch away from channels 5, 6 or 7 on your Wifi router.

**Reply by visualapproach** [reply]
Probably not related, but fun fact: a while ago I couldn't update my esp32/8266 modules over the air. And the tub controller was erratic and wouldn't load larger files like the pictures on the web pages. After a couple of days even the remote controlled garage doors and the weather station stopped working and I thought it might be because of a lot of wet snow covering the house. But I thought I'd investigate and brought my SDR dongle from a drawer. Scanned the usual frequencies and there was a steady and strong signal on one of the common remote control channels. Turned out one of the cables on my DIY ESP transmitter had come loose and it just went crazy loud on the aether. Plugged it in again and all stuff started working again.

**Reply by ChadH360** [reply]
Good shout - I'll see if I can find my SDR Dongle and see if there's anything in the air.
WiFi analyser on my mobile shows that channel 6 is pretty clear in the area, but then I think that will only show proper WiFi APs and not necessarily anything else on 2.4GHz.
But, it just seems quite coincidental that this was mentioned in issue 504:
> I put my WiFi transmit power back to auto and it still works but I notice a channel switch from 6 to 1 and it’s worked since. This does feel like a WiFi related issue. I’ll run more tests tomorrow

I wouldn't have even bothered with channels if it wasn't for that comment. I did notice that person was on Unifi as well.
As far as I'm aware, all other devices were working fine on Channel 6.

I'll let you know if I see anything crazy on the spectrum.

**Reply by visualapproach** [reply]
Check 433 and 868 MHz as well (or whatever is free band in your area). In my case the signal was on 433 but so strong it was disturbing 2.4GHz

**Reply by metaljay** [reply]
Just to add, I also have UniFi and occasionally unable to access the web interface yet MQTT still works fine. I will play around with the channel as I’ve typically have rebooted the pump for it to work again

**Comment by ChadH360** [comment]
> I will play around with the channel as I’ve typically have rebooted the pump for it to work again

How did you get on with channel adjustments? Did you see the same issue as me?

**Comment by Joho67** [comment]
I had this exact issue.  Changed router to ch8 and problem went away.
